modules/oe_subscriptions_digest/tests/src/Kernel/MailFlagtest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_subscriptions_digest\Kernel;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\Tests\flag\Traits\FlagCreateTrait;
+
+/**
+ * Class to test mail flag.
+ */
+class MailFlagtest extends KernelTestBase {
+
+  use FlagCreateTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'flag',
+    'message_subscribe_email',
+    'node',
+    'oe_subscriptions',
+    'system',
+    'user',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->installEntitySchema('flagging');
+    $this->installConfig([
+      'message_subscribe_email',
+    ]);
+  }
+
+  /**
+   * Tests the flag for mails are created from subscribe flag.
+   */
+  public function testFlagMirroring(): void {
+    // Create a flag with wrong prefix.
+    $flagStorage = $this->container->get('entity_type.manager')->getStorage('flag');
+    $this->createFlagFromArray(['id' => 'another_flag']);
+    $this->assertNull($flagStorage->load('email_something'));
```

Comment:
Good test, but let's load all flags to make the assertion. With the current test, if the code is changed and the flag created starts with `mail_`, it wouldn't fail.